pennylane/beta/plugins/default_tensor.py

```diff
@@ -447,40 +491,60 @@ def apply(self, operation, wires, par):
                         self.num_wires
                     )
                 )
-            state_node = self._create_basis_state(par[0], wires)
-            self._state_node.tensor = self._asarray(state_node, dtype=self.C_DTYPE)
+            state_tensor = np.zeros(tuple([2] * len(wires)))
+            state_tensor[tuple(par[0])] = 1
+            tensor = self._array(state_tensor, dtype=self.C_DTYPE)
+            self._clear_network()
+            self._add_initial_state_nodes([tensor], [full_wires_list], ["BasisState"])
             return
 
         A = self._get_operator_matrix(operation, par)
-        num_mult_idxs = len(wires)
-        A = self._reshape(A, [2] * num_mult_idxs * 2)
+        num_wires = len(wires)
+        A = self._reshape(A, [2] * num_wires * 2)
         op_node = self._add_node(A, wires=wires, name=operation)
-        for idx, w in enumerate(wires):
-            self._add_edge(op_node, num_mult_idxs + idx, self._state_node, w)
-            self._free_edges[w] = op_node[idx]
-        # TODO: can be smarter here about collecting contractions?
-        self._state_node = tn.contract_between(
-            op_node, self._state_node, output_edge_order=self._free_edges
-        )
 
-    def create_nodes_from_tensors(self, tensors: list, wires: list, observable_names: list):
-        """Helper function for creating tensornetwork nodes based on tensors.
+        if self._rep == "exact":
+            for idx, w in enumerate(wires):
+                tn.connect(op_node[num_wires + idx], self._terminal_edges[w])
+                self._terminal_edges[w] = op_node[idx]
+        elif self._rep == "mps":
+            raise NotImplementedError
+            # if len(wires) == 1:
+            #    self.mps.apply_one_site_gate(op_node, wires[0])
+            # elif len(wires) == 2:
+            #    if abs(wires[1]-wires[0]) == 1:
+            #        ret = self.mps.apply_two_site_gate(op_node, *wires)
+            #        # TODO: determine what ``ret`` is and if it is useful for anything
+            #        # TODO: pass ``max_singular_values`` or ``max_truncation_error``
+            #    else:
+            #        # only nearest-neighbours are natively supported
+            #        print("ruh roh")
+            # else:
+            #    raise NotImplementedError
+
+    def create_nodes_from_tensors(self, tensors, wires, observable_names, key):
+        """Helper function for creating TensorNetwork nodes based on tensors.
 
         Args:
-          tensors (np.ndarray, tf.Tensor, torch.Tensor): tensors of the observables
+          tensors (Sequence[np.ndarray, tf.Tensor, torch.Tensor]): tensors of the observables
           wires (Sequence[Sequence[int]]): measured subsystems for each observable
-          observable_names (Sequence[str]): name of the operation/observable
+          observable_names (Sequence[str]): names of the operation/observable
```

Comment:
Here the argument types are listed as `Sequence`, yet above they are listed as `list`.